icu4c/source/common/rbbidata.cpp

```diff
@@ -110,17 +109,18 @@ void RBBIDataWrapper::init(const RBBIDataHeader *data, UErrorCode &status) {
         fReverseTable = (RBBIStateTable *)((char *)data + fHeader->fRTable);
     }
 
-    fTrie = utrie2_openFromSerialized(UTRIE2_16_VALUE_BITS,
-                                      (uint8_t *)data + fHeader->fTrie,
-                                      fHeader->fTrieLen,
-                                      NULL,           // *actual length
-                                      &status);
+    fTrie = ucptrie_openFromBinary(UCPTRIE_TYPE_FAST,
+                                   UCPTRIE_VALUE_BITS_ANY,
+                                   (uint8_t *)data + fHeader->fTrie,
+                                   fHeader->fTrieLen,
+                                   nullptr,           // *actual length
+                                   &status);
     if (U_FAILURE(status)) {
         return;
     }
 
```

Comment:
Please check that the ucptrie_getValueWidth(fTrie) is either UCPTRIE_VALUE_BITS_8 or UCPTRIE_VALUE_BITS_16 (not 32, and there could be more widths in the future). invalid-format-error if not. (good place to cache the is-8-bits boolean...)

Reply:
done